robot/server/server.go

```diff
@@ -72,24 +72,25 @@ func (s *Server) Tunnel(srv pb.RobotService_TunnelServer) error {
 
 	dialTimeout := defaultTunnelConnectionTimeout
 
-	// TODO(RSDK-5763): Start rejecting requests to unavailable ports once `app` has been
-	// updated to propagate `traffic_tunnel_endpoints` configs.
-	/*
-	   var destAllowed bool
-	  // Ensure destination port is available; otherwise error.
-	   for _, tte := range s.robot.ListTunnels() {
-	   if int(req.DestinationPort) == tte.Port {
-	   destAllowed = true
-	   if tte.ConnectionTimeout != 0 {
-	   dialTimeout = tte.ConnectionTimeout
-	  }
-	   break
-	  }
-	  }
-	   if !destAllowed {
-	  return fmt.Errorf("tunnel not available at port %d", req.DestinationPort)
-	  }
-	*/
+	// Ensure destination port is available; otherwise error.
+	var destAllowed bool
+	ttes, err := s.robot.ListTunnels(srv.Context())
+	if err != nil {
+		return err
+	}
+	for _, tte := range ttes {
+		if int(req.DestinationPort) == tte.Port {
+			destAllowed = true
+			if tte.ConnectionTimeout != 0 {
+				// Honor specified timeout if one exists (0 is use-default.)
+				dialTimeout = tte.ConnectionTimeout
+			}
+			break
+		}
+	}
+	if !destAllowed {
+		return fmt.Errorf("tunnel not available at port %d", req.DestinationPort)
+	}
```

Comment:
Re-requesting you @cheukt , since this PR now _also_ honors `traffic_tunnel_endpoints`.